src/stores/Store.ts

```diff
@@ -74,6 +78,184 @@ export default class Store {
+		}
+	};
+
+	@action selectMessage = (index: number) => {
+		this.setSelectedSession(JSON.parse(this.parsedMessagesHistory[index]).session);
+		this.setDictionaryName(JSON.parse(this.parsedMessagesHistory[index]).dictionary);
+		this.setSelectedMessageType(JSON.parse(this.parsedMessagesHistory[index]).messageType);
+		this.setEditorCode('{}');
+		this.setSendButtonTitle('Save');
+		this.setSendButtonIcon('');
+		this.setEditMessageMode(true);
+		this.setEditedMessageIndex(index);
+	};
+
+	@action setSendButtonIcon =(iconClass: string) => {
+		this.sendButtonIcon = iconClass;
+	};
+
+	@action setSendButtonTitle = (title: string) => {
+		this.sendButtonTitle = title;
+	};
+
+	@action setSelectedSession = (session: string | null) => {
+		this.selectedSession = session;
+		if (this.editMessageMode === false && session !== null) {
+			localStorage.setItem('selectedSession', session);
+		}
+	};
+
+	@action setSelectedMessageType = (messageType: string | null) => {
+		this.selectedMessageType = messageType;
+		if (this.editMessageMode === false && messageType !== null) {
+			localStorage.setItem('selectedMessageType', messageType);
+		}
+	};
+
+	@action setDictionaryName = (dictionary: string | null) => {
+		this.selectedDictionaryName = dictionary;
+		if (this.editMessageMode === false && dictionary !== null) {
+			localStorage.setItem('selectedDictionary', dictionary);
+		}
+	};
+
+	@action setEditorCode = (code: string) => {
+		this.editorCode = code;
+	};
+
+	@action getIndicatorByResult = (code: number): string => {
+		if (code === 200) return 'circleGreen';
+		return 'circleRed';
+	};
+
+    @action addIndicator = (indicatorClass: string) => {
+    	this.indicators.push(indicatorClass);
+    };
+
+    @action changeIndicator = (index: number, indicatorClass: string) => {
+    	this.indicators[index] = indicatorClass;
+    };
+
+	@action deleteIndicator = (index: number): string[] => {
+		const tmpArray: string[] = this.indicators.slice();
+		this.indicators = [];
+		tmpArray.forEach((item, i) => {
+			if (i !== index) {
+				this.addIndicator(item);
+			}
+		});
+		return this.indicators;
+	};
+
+	@action setFromFile = (flag: boolean) => {
+		this.fromFileMessage = flag;
+	};
+
+	@action addParsedMessage = (message: string, indicatorClass?: string) => {
+		this.addIndicator(indicatorClass || 'circleUnvisible');
+		this.parsedMessagesHistory.push(message);
+		localStorage.setItem('messageList', this.parsedMessagesHistory.toString());
+	};
+
+	@action clearParsedMessages = () => {
+		this.parsedMessagesHistory = [];
+		this.indicators = [];
+		localStorage.setItem('messageList', this.parsedMessagesHistory.toString());
+		if (this.editMessageMode === true) {
+			this.setEditMessageMode(false);
+		}
+	};
+
+	@action startApp = () => {
+		this.setSelectedSchemaType(localStorage.getItem('selectedSchemaType')
+			? localStorage.getItem('selectedSchemaType') as SchemaType : this.selectedSchemaType);
+		if (localStorage.getItem('session') === '0') {
```

Comment:
it's very hard to understand what is going on with
`localStorage.getItem('session') === '0'`
`localStorage.setItem('session', '1')`
on mount App you reset 'session' in localStorage, so why can't we just keep this in store, not in localStorage?